src/robotFactory.js

```diff
@@ -1,10 +1,74 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return 'Robot: ' + this.name + ', Chip version: '
+      + this.chipVersion + ', Weight: ' + this.weight;
+  }
```

Comment:
I think you didn't fully understand @naziamoff comment about consistency.

```suggestion
  getInfo() {
    return `Robot: ${this.name}, ` 
    + `Chip version: ${this.chipVersion}, `
    + `Weight: ${this.weight}`;
  }
```

Reply:
Ohh, now I get it! I just never use it like this... Thank you, guys!